pkg/util/kubernetes/autoscalers/processor.go

```diff
@@ -146,13 +151,25 @@ func (p *Processor) ProcessWPAs(wpa *v1alpha1.WatermarkPodAutoscaler) map[string
 }
 
 func makeChunks(batch []string) (chunks [][]string) {
-	for i := 0; i < len(batch); i += chunkSize {
-		if i+chunkSize > len(batch) {
-			chunks = append(chunks, batch[i:])
-			break
+	// uriLength is used to avoid making a query that goes beyond the maximum URI size.
+	var uriLength int
+	var tempBucket []string
+
+	for _, val := range batch {
+		// Length of the query plus comma, time and space aggregators that come later on.
+		tempSize := len(url.PathEscape(val)) + extraQueryCharacters
+		uriLength = uriLength + tempSize
+		if uriLength >= maxCharactersPerChunk || len(tempBucket) >= chunkSize {
+			// The metric name can be at maximum 200 characters. Kubernetes limits the labels to 63 characters.
+			// Autoscalers with enough labels to form single a query of more than 7k characters are not supported.
```

Comment:
can we log an error if it appends?

Reply:
I can log in debug, it's not really error level in my opinion.

Reply:
I was talking about the case that is not supported: if one unique query generate a URI to long